### DISCUSSION TITLE
Latest tag behavior

### DISCUSSION THREAD

**Discussion by hovav019** [verdaccio/verdaccio#3838] [Q&A] [answered]
Hi,

The verdacio behavior while uploading the package is that the latest package deployed is the latest.
Is there any option to change this behavior to work according to the package versions?

**Comment by juanpicado** [comment]
Could you elaborate l? I don't get it

**Comment by hovav019** [comment]
![photo_2023-06-11_07-59-53](https://github.com/verdaccio/verdaccio/assets/84805111/18887af3-e9e1-493c-a282-06cc23cf96e7)

As you can see in the photo attached I have uploaded a version of the react package for the server.

I have uploaded version 16.4.0 of the package and it set this version to latest instead of the higher version

**Reply by juanpicado** [reply]
I see, have you consider using eg: `npm publish --tag beta` on publish? If you no indicated the tag package managers and verdaccio asume is latest

**Reply by hovav019** [reply]
‎‏I want to use verdacio as a private npm registry because we are developing on a closed environment without any connection to the internet.
So uploading packages as beta won't solve the problem.

**Reply by juanpicado** [reply]
Sorry I don't manage to  follow what you mean, you might need to expand with examples to be more clear.

**Reply by hovav019** [reply]
We would like the verdaccio default behavior to be changed so that when we upload an existing package it will figure out what the highest version is and update the latest tag accordingly. 
Because we are working in an air-gapped environment we usually perform mass publish for many packages, depending on the project. Doing so we can't check for every package if it is existing already and what is the current highest version in order to add the --tag to the publish command

**Reply by mbtools** [reply]
@hovav019 verdaccio works the same way as npm: The last package you publish is tagged with "latest". 

Here is an example where I published 1.0.0 first, then 0.0.9:

![image](https://github.com/verdaccio/verdaccio/assets/59966492/11eaf495-aa45-4f71-962c-574348d24fd2)

**Reply by juanpicado** [reply, marked as answer]
> We would like the verdaccio default behavior to be changed so that when we upload an existing package it will figure out what the highest version is and update the latest tag accordingly. Because we are working in an air-gapped environment we usually perform mass publish for many packages, depending on the project. Doing so we can't check for every package if it is existing already and what is the current highest version in order to add the --tag to the publish command

I understand what you are trying to do but I don't think this is going to happen in verdaccio, sounds to me like a custom solution in your side, verdaccio is just an api, if you want to know which version is the latest just ask for the metadata (eg: https://registry.npmjs.org/jquery, is the same that `npm show` does under the hood) and call publish with or without `--tag` accordingly, that's all I can suggest.
